google-cloud-spanner/src/test/java/com/google/cloud/spanner/SessionPoolTest.java

```diff
@@ -1614,15 +1614,16 @@ public void testSessionNotFoundPartitionedUpdate() {
     assertThat(impl.executePartitionedUpdate(statement)).isEqualTo(1L);
   }
 
+  @SuppressWarnings("rawtypes")
   @Test
   public void testSessionMetrics() throws Exception {
     // Create a session pool with max 2 session and a low timeout for waiting for a session.
     options =
         SessionPoolOptions.newBuilder()
             .setMinSessions(1)
             .setMaxSessions(2)
-            .setMaxIdleSessions(0)
             .setInitialWaitForSessionTimeoutMillis(50L)
+            .setAcquireSessionTimeout(null)
```

Comment:
If we don't set this, then does it add a default timeout of 60 secs during acquiring a session?
Since this test validates the opencensus metrics I was thinking why do we need to add this?

Reply:
Yeah, this part is confusing, as we have made multiple changes to it through the years:
1. The `acquireSessionTimeout` is a 'real' timeout, as in: It will throw an error if it is exceeded.
2. In addition, we have the `InitialWaitForSessionTimeout` which is the original implementation that we had, and also what this test verifies. It does not throw an error if it is exceeded. Instead, it only adds an annotation to the trace.

So the reason that we are setting `acquireSessionTimeout` to null is to verify the correct behavior of the field `InitialWaitForSession`, as that field is ignored if there is an `acquireSessionTimeout`. See this line: https://github.com/googleapis/java-spanner/blob/ee477c2e7c509ce4b7c43da3b68c1433c59e46fb/google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java#L1687